pkg/git/gitproviderfactory/gitproviderfactory.go

```diff
@@ -17,22 +17,22 @@ limitations under the License.
 package gitproviderfactory
 
 import (
+	"encoding/base64"
 	"fmt"
-	"strconv"
-	"strings"
-
 	"github.com/redhat-appstudio/application-service/gitops"
 	"github.com/redhat-appstudio/build-service/pkg/boerrors"
 	"github.com/redhat-appstudio/build-service/pkg/git/github"
 	"github.com/redhat-appstudio/build-service/pkg/git/gitlab"
 	"github.com/redhat-appstudio/build-service/pkg/git/gitprovider"
+	corev1 "k8s.io/api/core/v1"
```

Comment:
Can we avoid bringing k8s dependencies into git packages?

Reply:
is that a big concern? Im afraid i cannot avoid that w/out adding some degree of mess :(

Reply:
@mshaposhnik, the idea is to have git packages independent from the main code, so they can be reused from other places like GitHub actions. While creating a map is not a big deal, building k8s secret object and kubernetes dependencies looks like overhead.
I think we can easily remove that dependency, by just passing a map with credentials as it was done before. All the validations are already happened, so we can be sure we have a valid structure and can easily guess auth type based on the fields in the map.

Reply:
ok, i refactored it